File: 75-sort-colors/sort-colors.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var sortColors = function (nums) {
    let n = nums.length;
    let low = 0;
    let mid = 0;
    let high = n - 1;

    while (mid <= high) {
        if (nums[mid] == 0) {
            let temp = nums[mid]
            nums[mid] = nums[low]
            nums[low] = temp

            low++;
            mid++
        } else if (nums[mid] == 1) {
            mid++;
        } else if (nums[mid] == 2) {
            let temp = nums[mid]
            nums[mid] = nums[high]
            nums[high] = temp

            high--;
        }
    }
    return nums
};

File: 75-sort-colors/sort-colors.java
class Solution {
    public void swap(int a, int b,int[] nums) {
        int temp = nums[a];
        nums[a] = nums[b];
        nums[b] = temp;
    }

    public void sortColors(int[] nums) {
        int n = nums.length;
        int low = 0;
        int mid = 0;
        int high = n - 1;

        while (mid <= high) {
            if (nums[mid] == 0) {
                swap(low, mid, nums);
                low++;
                mid++;
            } else if (nums[mid] == 2) {
                swap(high, mid, nums);
                high--;
            } else {
                mid++;
            }
        }
    }
}